mobile/lib/services/backup.service.dart

```diff
@@ -525,6 +529,81 @@ class BackupService {
         AssetType.video => "VIDEO",
         AssetType.other => "OTHER",
       };
+
+  Future<File?> _processWhatsAppMedia(File file, String fileName, DateTime creationDate, AssetEntity asset) async {
+    // Heuristics to determine if the file is likely a WhatsApp file:
+    // - File name matches a UUID pattern.
+    // - File has no GPS location data.
+    // - File has no device information (e.g., no camera or device ID).
+    // - File is imported (not captured directly by the device's camera).
+    final isUuidName = _isUuidFileName(fileName);
+    final noLocation = (asset.latitude == 0.0 && asset.longitude == 0.0);
+    final noDeviceInfo = asset.deviceId == null || asset.deviceId!.isEmpty;
+    final isImported = asset.type != AssetType.image && asset.type != AssetType.video;
+
+    final isLikelyWhatsAppFile = isUuidName && noLocation && noDeviceInfo && isImported;
+
+    if (!isLikelyWhatsAppFile) {
+      debugPrint("File does not meet WhatsApp criteria: $fileName");
+      return file;
+    }
+
+    try {
+      final fileExtension = fileName.split('.').last.toLowerCase();
+
+      if (['jpg', 'jpeg', 'png'].contains(fileExtension)) {
+        final imageBytes = await file.readAsBytes();
+        final image = img.decodeImage(imageBytes);
+
+        if (image == null) {
+          return file;
+        }
+
+        final exifData = img.ExifData();
+        exifData.dateTimeOriginal = creationDate;
+
+        final updatedImageBytes = img.encodeJpg(image, exif: exifData);
```

Comment:
Hello, we will not accept PRs that edit the source image as this greatly increases the chance of data loss. If you want to pursue this project, please join our Discord and discuss in the #contributing channel before working on the next PR.